### PR TITLE
Generator implementation for V2 smoke tests for C2J files.

### PR DESCRIPTION
**Pull request by AlexDaines**
## Description
Implements the generator for smoke tests v2 format based on the new smoke-2.json schema. Key features:

- Supports all configuration options from the smoke-2.json schema
- Handles complex input types including arrays and nested objects
- Generates both success and failure test cases
- Includes proper test categorization
- Manages special cases for Tags and Includes objects

Questions for review:
1. Special case handling for Tags/Includes - should this be more generic?
2. Configuration defaults - should we explicitly set schema defaults?
3. Error handling - any additional checks needed beyond ErrorCode?
4. Input validation - should we validate against operation parameters?
5. Test categorization - any additional AWS-specific categories needed?

## Motivation and Context
This implementation supports the new smoke tests format (smoke-2.json) which provides more configuration options and better structure for test cases. The generator creates test files that:
- Allow per-test configuration
- Support all endpoint configuration options
- Handle complex input structures
- Provide proper error handling and validation

## Testing
- Verified generation with ACM service model
- Tested various scenarios including:
  - Success and failure cases
  - Complex input structures (arrays, nested objects)
  - Special cases (Tags, Includes)
  - Edge cases (null values, empty strings)
  - All configuration options
- Generated tests compile successfully

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Review by normj — CHANGES_REQUESTED** [review]

File: generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt [hunk 1]
```diff
@@ -8,6 +8,7 @@
 	AddLicenseHeader();
 #>
 using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (dscpinheiro) [inline generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt:13]: (I'm not reviewing the PR, but...) Is there a reason why you're using MSTest as the test framework? 

The [hand-written files](https://github.com/aws/aws-sdk-net/blob/main/sdk/test/Services/SecretsManager/SmokeTests/SmokeTest.cs) we have right now use xUnit (meaning they can run in parallel) and using xUnit will allow you to test your PR without making any changes to our build system (as long as they're found by the projects in this folder: https://github.com/aws/aws-sdk-net/tree/v4-development/sdk/test/SmokeTests).

File: generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt [hunk 2]
```diff
@@ -26,36 +27,223 @@ namespace AWSSDK_DotNet.IntegrationTests.SmokeTestsV2
...
+            {
+                foreach (string key in input.PropertyNames)
+                {
+                    var value = input[key];
+                    if (value != null)
+                    {
+                        if (value.IsString)
+                        {
+#>
+            request.<#=key#> = "<#=value.ToString()#>";
```
[the remark this answers is not in the document]
Comment (AlexDaines) [reply generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt]: Now using FindPropertyName, which should handles cases where the JSON key name doesn't exactly match the .NET property name.

File: generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt [hunk 3]
```diff
@@ -26,36 +27,223 @@ namespace AWSSDK_DotNet.IntegrationTests.SmokeTestsV2
...
+                var errorId = GetExpectedErrorId(testCase);
+#>
+            try
+            {
+                var response = await serviceClient.<#=modeledOperation.Name#>Async(request);
+                Assert.Fail("Expected <#=errorId ?? "an error response"#> exception, but the call succeeded.");
+            }
+            catch (Amazon<#=this.Config.ClassName#>Exception ex)
+            {
+                Assert.AreEqual("<#=errorId#>", ex.ErrorCode);
```
[the remark this answers is not in the document]
Comment (AlexDaines) [reply generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt:296]: Null check for this added

File: generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt (hunk 1 above)
Comment (AlexDaines) [reply generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt:13]: @normj

**Comment by peterrsongg** [comment]
I tried running the generator and it seems like this needs to be changed as well

```
            if (this.Configuration.ServiceModel.HasSmokeTestsV2)
            {
                ExecuteIntegrationTestGenerator(new SmokeTestsV2(), "SmokeTestsV2.cs");
            }
```

with this logic, each file will be `SmokeTestsV2.cs` and one will overwrite the other. You should probably do something similar to the other generators and do something like

```
ExecuteIntegrationTestGenerator(new SmokeTestsV2(), serviceName + "SmokeTestsV2.cs");
```

**Comment by peterrsongg** [comment]
^^ looking into the problem I mentioned above a bit more, it seems like it thinks these generated integ tests are orphaned files because they are not a part of the `generatedTestFiles` that get populated here
https://github.com/aws/aws-sdk-net/blob/main/generator/ServiceClientGenerator/Program.cs#L64-L79

since it is not part of that list, when it is passed into `GeneratorDriver.RemoveOrphanedShapesAndServices` it gets removed. This is probably the first instance of a generated integ tests so you should add these generated integ tests to the list so it isn't removed

File: generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt [hunk 4]
```diff
@@ -26,36 +27,283 @@ namespace AWSSDK_DotNet.IntegrationTests.SmokeTestsV2
...
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (IsSuccessExpected(testCase))
+            {
+#>
+            var response = await serviceClient.<#=modeledOperation.Name#>Async(request);
```
Comment (peterrsongg) [inline generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt:278]: Add `.ConfigureAwait(false)` at the end of this async call. We should always be adding that at the end of async calls, unless you really know what you're doing. See https://devblogs.microsoft.com/dotnet/configureawait-faq/#why-would-i-want-to-use-configureawait(false) for why we should use `configureAwait(false)`

File: generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt [hunk 5]
```diff
@@ -26,36 +27,283 @@ namespace AWSSDK_DotNet.IntegrationTests.SmokeTestsV2
...
+<#
+            }
+            else
+            {
+                var errorId = GetExpectedErrorId(testCase);
+#>
+            try
+            {
+                var response = await serviceClient.<#=modeledOperation.Name#>Async(request);
```
Comment (peterrsongg) [inline generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt:287]: same comment as above

File: generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt (hunk 1 above)
Comment (peterrsongg) [reply generator/ServiceClientGeneratorLib/Generators/TestFiles/SmokeTestsV2.tt:13]: unless you want to make a buildsystem change I'd probably listen to daniel here

**Review by peterrsongg — CHANGES_REQUESTED** [review]

**Comment by dscpinheiro** [comment]
Closing as the smoke tests work is on hold. Will revisit this later.
